Add initial models - fixes #16

rebbrunner opened the pull request:
Adds initial support for #16 - create models for project selection.

IDs are automatically generated by Django for every model instance.
Lists are implemented using OneToMany relationships.

rebbrunner committed:
Add initial models - fixes #16

rebbrunner committed:
Add new models to admin site - fixes #16

psybers commented on website/models.py:
```diff
@@ -68,6 +68,56 @@ class Meta:
         )
 
+class Dataset(models.Model):
+    #unique ids are automatically generated by django for every model instance
```
we dont need this comment here, since it is standard django practice

psybers commented on website/models.py:
```diff
@@ -68,6 +68,56 @@ class Meta:
+        return self.input_dataset + ' - ' + self.input_selection
+
+class Filter(models.Model):
+    project_selector = models.ForeignKey(ProjectSelector, on_delete=models.CASCADE)
```
Isnt this backwards?  A filter can be used by many different project selectors.  This is a many-many relation.

psybers replied:
This is still backwards. ProjectSelector picks Filters. This is the other way.

psybers commented on website/models.py:
```diff
@@ -68,6 +68,56 @@ class Meta:
+        return self.name
+
+class ProjectSelector(models.Model):
+    input_dataset = models.IntegerField(blank=True)
```
Most of the 'ints' in the issue description of the models were foreign keys.  Is this how Django models foreign keys?

rebbrunner committed:
Remove comment, change to many to many, update ints to foreign keys

psybers commented on website/models.py:
```diff
@@ -68,6 +68,52 @@ class Meta:
+class ProjectSelector(models.Model):
+    input_dataset = models.ForeignKey(Dataset, blank=True, null=True)
+    input_selection = models.ForeignKey(Selection, blank=True, null=True)
+    output = models.IntegerField()
```
lets rename this `output_selection` and it should be a foreign key to Selection

rebbrunner committed:
Change output in ProjectSelector to ForeignKey

psybers commented:
project_selector, project_transformer, and analysis probably need foreign keys to a User

rebbrunner committed:
Add user field, update names

Change ManyToMany field names to plural,
Add foreign key to user for project selector, transformer, and analysis.
